pyblish/logic.py

```diff
@@ -354,11 +355,13 @@ def Iterator(plugins, context, state=None):
         "ordersWithError": set()
     }
 
-    # We'll add "default" target if no targets are registered. This happens
-    # when running the Iterator directly without registering any targets.
-    targets = registered_targets() or ["default"]
+    # Include "default" target when no targets are requested.
+    _targets = targets or ["default"]
 
-    plugins = plugins_by_targets(plugins, targets)
+    if targets is None:
+        _targets += registered_targets()
```

Comment:
This is interesting. Did anything change here? Under the two possible circumstances of either (a) passing `targets` as an arguments or (b) not passing it, in which case it's `None`, it seems the result is the same?

Reply:
Not sure what you mean, `targets` are use by default if passed explicitly, otherwise "default" and registered targets would be used. Previously, registered targets were used by default, otherwise "default" target would be used. 

As [pyblish.util._convenience_iter](https://github.com/pyblish/pyblish-base/blob/acf97ef9fdc2fc506daac7817db99073a79a7c56/pyblish/util.py#L98) was automatically registering the "default" target when no explicit targets were passed, I didn't keep the exclusive condition between registered targets and "default" target.

I updated the logic to make it clearer:
```python
if not targets:
    targets = ["default"] + registered_targets()
```

Reply:
Oh, yes. You're right. Sorry I'm not sure what I meant either. xD